api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -232,6 +232,41 @@ message RemoteJwks {
   // Duration after which the cached JWKS should be expired. If not specified, default cache
   // duration is 5 minutes.
   google.protobuf.Duration cache_duration = 2;
+
+  // Fetch Jwks asynchronously in the main thread before the listener is activated.
+  // Fetched Jwks can be used by all worker threads.
+  //
+  // If this feature is not enabled:
+  //
+  // * The Jwks is fetched on-demand when the requests come. During the fetching, first
+  //   few requests are paused until the Jwks is fetched.
+  // * Each worker thread fetches its own Jwks since Jwks cache is per worker thread.
+  //
+  // If this feature is enabled:
+  //
+  // * Fetched Jwks is done in the main thread before the listener is activated. Its fetched
+  //   Jwks can be used by all worker threads. Each worker thread doesn't need to fetch its own.
+  // * Jwks is ready when the requests come, not need to wait for the Jwks fetching.
+  //
+  JwksAsyncFetch async_fetch = 3;
+}
+
+// Fetch Jwks asynchronously in the main thread when the filter config is parsed.
+// The listener is activated only after the Jwks is fetched.
+// When the Jwks is expired in the cache, it is fetched again in the main thread.
+// The fetched Jwks from the main thread can be used by all worker threads.
+message JwksAsyncFetch {
+  // If true, enable this feature.
+  bool enable = 1;
+
+  // If false, the listener is activated after the initial fetch is completed.
+  // If true, it is activated without waiting for the initial fetch to complete.
+  // Default is false.
+  bool fast_listener = 2;
+
+  // The number of retries when Jwks fetch fails with the networking error.
+  // Default is no retry.
+  uint32 network_fail_retries = 3;
```

Comment:
Shall we reuse normal HTTP RetryPolicy here, and make HTTP Async Client aware of that?

Reply:
Good idea.  I will add core.Retry config as part of RemoteJwks, Change  common/JwksFetcher to support it.
That can be done as separate pr.  The issue is tracked [here](https://github.com/envoyproxy/envoy/issues/16319)

In this pr, I will remove retry